src/MongoDB.Driver/MongoDatabaseImpl.cs

```diff
@@ -452,11 +452,12 @@ public override IMongoDatabase WithWriteConcern(WriteConcern writeConcern)
         // private methods
         private AggregateOperation<TResult> CreateAggregateOperation<TResult>(RenderedPipelineDefinition<TResult> renderedPipeline, AggregateOptions options)
         {
+            var messageEncoderSettings = GetMessageEncoderSettings();
```

Comment:
Some of the spec tests expect that `aggregation` operation also will use encryption. I've added encryption for all methods which use `_messageEncoderSettings` in this class based on the json tests requirements. Except for `CreateAggregateToCollectionFindOperation`, since there is no tests for this operation. Probably we can update `_messageEncoderSettings` for this method too. And if so, we can update the `MongoDatabaseImpl` in similar to `MongoCollectionImpl` way. I mean having updated `messageEncoderSettings` in the constructor.

Reply:
Done